.github/workflows/python-ci.yml

```diff
@@ -12,39 +12,39 @@ on:
 concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
-  
+
 jobs:
   ci:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         os: [ ubuntu-latest, windows-latest, macos-latest ]
-        python-version: [ "3.12.1" ]
     steps:
-      - uses: actions/checkout@v4
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v4
-        with:
-          python-version: ${{ matrix.python-version }}
-      - uses: abatilo/actions-poetry@v3.0.1
+      - name: "Check out repository"
+        uses: actions/checkout@v4
+
+      - name: "Install uv"
+        uses: astral-sh/setup-uv@v5
         with:
-          poetry-version: 1.7.1
-      - name: Cache Poetry
-        uses: actions/cache@v4
+          # Install a specific version of uv.
+          version: "0.5.25"
+
+      - name: "Set up Python"
+        uses: actions/setup-python@v5
```

Comment:
This action should cache dependencies, does it support uv?

Reply:
The uv action supports caching: https://docs.astral.sh/uv/guides/integration/github/#caching I see no way to invalidate the cache both by changes in `uv.lock` and `.python-version`, except with the manual cache management option.